src/OpenLoco/src/World/CompanyAi.cpp

```diff
@@ -564,39 +629,39 @@ namespace OpenLoco
     {
         company.var_85C3 &= ~((1U << 0) | (1U << 1));
 
-        if (kThoughtTypeFlags[enumValue(thought.type)] & ((1U << 15) | (1U << 16)))
+        if (thoughtTypeHasFlags(thought.type, ThoughtTypeFlags::airBased | ThoughtTypeFlags::waterBased))
         {
             return true;
         }
 
-        auto findRequiredUnk = [&thought, &company]() -> int8_t {
-            for (auto i = 0U; i < thought.var_03; ++i)
+        auto findRequiredAiStation = [&thought, &company]() -> int8_t {
```

Comment:
If we had a helper function that returns a span for stations we could turn this into a predicate and use std::find_if instead.

Reply:
I will come back to this later if i can confirm that stations is always the numStations size.